oracle/src/oraclewizard.cpp

```diff
@@ -454,8 +458,10 @@ void LoadSetsPage::readSetsFromByteArray(QByteArray data)
             zipDownloadFailed(tr("Xz extraction failed."));
             return;
         }
+        const auto &outBufferData = outBuffer->data();
 
-        future = QtConcurrent::run(wizard()->importer, &OracleImporter::readSetsFromByteArray, outBuffer->data());
+        future = QtConcurrent::run(
+            [this, &outBufferData] { return wizard()->importer->readSetsFromByteArray(outBufferData); });
```

Comment:
handing this by reference to a different thread doesn't look very safe at all

Reply:
The pointer wasn't safe and led to access errors, but as an object it seems fine from my testing

Reply:
does this duplicate everything then?

Reply:
Not sure what you mean

Reply:
it's oracle, we shouldn't put too much effort in it imo